src/modes/draw_polygon.js

```diff
@@ -0,0 +1,91 @@
+var {isEnterKey, isEscapeKey} = require('../lib/common_selectors');
+var Polygon = require('../feature_types/polygon');
+
+module.exports = function(ctx) {
+
+  var geojson = {
+    "type": "Feature",
+    "properties": {},
+    "geometry": {
+      "type": "Polygon",
+      "coordinates": [[[0,0], [0, 0],[0, 0]]]
+    }
+  }
+
+  var feature = new Polygon(ctx, geojson);
+
+  var stopDrawingAndRemove = function() {
+    ctx.events.changeMode('default');
+    ctx.store.delete(feature.id);
+  }
+
+  var pos = 0;
+
+  var onMouseMove = function(e) {
+    if(pos === 0) {
+      feature.updateCoordinate(`0.${0}`, e.lngLat.lng, e.lngLat.lat);
+      feature.updateCoordinate(`0.${1}`, e.lngLat.lng, e.lngLat.lat);
+    }
+    else {
+      feature.updateCoordinate(`0.${pos}`, e.lngLat.lng, e.lngLat.lat);
+    }
+  }
+
+  var onClick = function(e) {
+    // did we click on the last point
+    // did we click on the first point
+    pos++;
+  }
+
+  var onFinish = function(e) {
+    feature.removeCoordinate(`0.${pos}`);
+    pos--;
+    if(pos < 2) {
+      stopDrawingAndRemove();
+    }
+    else {
+      ctx.events.changeMode('default');
+    }
+  }
+
+  return {
+    start: function() {
+      ctx.ui.setClass('mapbox-gl-draw_mouse-add');
+      this.on('mousemove', () => true, onMouseMove);
+      this.on('click', () => true, onClick);
```

Comment:
If they return false the next function does not run. If they return true the next function does run and no other event handler is tested.